lib/xfstests_utils.pm

```diff
@@ -601,12 +601,17 @@ sub test_run_without_heartbeat {
         sleep 2;
         copy_all_log($category, $num, $fstype, $btrfs_dump, $raw_dump, $scratch_dev, $scratch_dev_pool, 1);
 
-        prepare_system_shutdown;
-        reset_consoles if check_var('DESKTOP', 'textmode');
-        ($virtio_console == 1) ? power_action('reboot') : send_key 'alt-sysrq-b';
-        reconnect_mgmt_console if is_pvm;
-        check_var('DESKTOP', 'textmode') ? $self->wait_boot_textmode : $self->wait_boot;
-        is_public_cloud() ? select_console('root-console') : select_serial_terminal();
+        if (is_public_cloud()) {
+            $cloud_instance->softreboot(timeout => get_var('PUBLIC_CLOUD_REBOOT_TIMEOUT', 600));
+        }
+        else {
+            prepare_system_shutdown;
+            reset_consoles if check_var('DESKTOP', 'textmode');
+            ($virtio_console == 1) ? power_action('reboot') : send_key 'alt-sysrq-b';
+            reconnect_mgmt_console if is_pvm;
+            check_var('DESKTOP', 'textmode') ? $self->wait_boot_textmode : $self->wait_boot;
+            is_public_cloud() ? select_console('root-console') : select_serial_terminal();
```

Comment:
```suggestion
            select_serial_terminal();
```

and you need to move `select_console('root-console')` inside `if`

Reply:
or maybe just drop it :thinking:  anyway this ternary  operator does not make sense anymore because you already inside `is_public_cloud()== false`

Reply:
Good catch, thanks. That was a leftover of a conflict resolution due to rebasing after https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/20272

Reply:
Fixed in https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/20422/commits/dad6110131b9f99c709a7bec0ef654ac2b3a6f5a